### ISSUE TITLE
[MultiKueue] Report a ClusterQueue as inactive (misconfigured) if there is ProvReq used with MK

### ISSUE THREAD

**Issue by mimowo** [kubernetes-sigs/kueue#2023]
**What would you like to be added**:

Validation for ClusterQueue, if there is a MK and ProvReq admission check configured.

**Why is this needed**:

Provisioning nodes on the management cluster does not make sense.
We want to fail fast, and warn user about possibly wasted money to scale-up the cluster.

**Proposed approach**:

Use a mechanism similar to the one here: https://github.com/kubernetes-sigs/kueue/pull/1635.

**Comment by mimowo** [comment]
I reviewed https://github.com/kubernetes-sigs/kueue/pull/2047, and I think we could follow the pattern here. 

The `AdmissionCheck` condition would be `CompatibleWithMultiKueue`, and the reason for inactive ClusterQueue could be `AdmissionCheckNonCompatibleWithMultiKueue`. We would do the check inside `updateWithAdmissionChecks` as for other checks.

**Comment by alculquicondor** [comment]
The only problem is that the condition would be specific to MultiKueue. What if other checks need similar semantics against others?

I would rather sit on this one for now until we observe more admission checks, in-tree or out-of-tree.

**Comment by mimowo** [comment]
> What if other checks need similar semantics against others?

Right, this approach cannot be used for arbitrary pairs of admission checks. However, MultiKueue seems more than an admission check. For example, it has a global configuration in the config map [link](https://github.com/kubernetes-sigs/kueue/blob/de1fe1f6ff69e88e62c51f89428a5addc00ffadc/apis/config/v1beta1/configuration_types.go#L71-L72).

> I would rather sit on this one for now until we observe more admission checks, in-tree or out-of-tree.

I see, but it can take a long time until we have other pairs of AdmissionChecks which don't like each other, and having some protection before graduating MK and ProvReq to Beta would be nice. 

The approach using the existing mechanism should be very quick to implement, and if one day we have a more generic mechanism, developed for the needs of other AC pairs, then we could switch to it.

**Comment by alculquicondor** [comment]
Let's wait and see

**Comment by alculquicondor** [comment]
@mimowo I think we don't have a proper design for this. And it hasn't proved to be very useful. Should we close it?

**Comment by mimowo** [comment]
I'm ok to close it until we revisit the design or some evidence for users running into this issue.

**Comment by mimowo** [comment]
/reopen
I believe with the recent changes (https://github.com/kubernetes-sigs/kueue/pull/3254) to make cache aware of the MultiKueue and ProvisioningRequest AdmissionChecks we can easily validate this conditions.
cc @mbobrovskyi @mszadkow

**Comment by samos123** [comment]
Hitting this and unsure why this is happening. Any insights?

Kubectl describe workload
```
Status:
  Conditions:
    Last Transition Time:  2025-06-16T21:48:02Z
    Message:               ClusterQueue cluster-queue is inactive
    Observed Generation:   1
    Reason:                Inadmissible
    Status:                False
    Type:                  QuotaReserved
```


```
k get clusterqueue
NAME            COHORT   PENDING WORKLOADS
cluster-queue            31

k get localqueue
NAME               CLUSTERQUEUE    PENDING WORKLOADS   ADMITTED WORKLOADS
multislice-queue   cluster-queue   31                  0
```

**Comment by mimowo** [comment]
@samos123 what is your CQ configuration? Maybe provide the entire `kubectl describe` for the CQ

**Comment by mimowo** [comment]
/close
Let me reopen in the new context
